habitat-lab/habitat/sims/habitat_simulator/actions.py

```diff
@@ -197,8 +197,25 @@ def get(self):
 @registry.register_action_space_configuration(name="velocitycontrol")
 class HabitatSimVelocityCtrlActionSpaceConfiguration(ActionSpaceConfiguration):
     def get(self):
+        if not HabitatSimActions.has_action("VELOCITY_CTRL"):
+            HabitatSimActions.extend_action_space("VELOCITY_CTRL")
         return {
             HabitatSimActions.VELOCITY_CTRL: habitat_sim.ActionSpec(
                 "velocity_control"
             ),
+            # The perfect actions are needed for the oracle planner
```

Comment:
Is there a way to avoid it? Where exactly it is used in the code?

Reply:
The shortest path agent these actions to be available to plan the path. But it might be okay to stop supporting that with this new version of the navigation tasks. The shortest path agent can only be used by 1 of the three new action spaces anyway. That action space is discrete_waypoint_controller which is trying to replicate the old habitat 1.0 actions. If anyone wants to use the shortest path planner, they could just use the older action space and get the outputs from there.

Let me know what you think?